File: MinimumSizeSubarraySum/MinimumSizeSubarraySum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int start = 0, end = 0, S = 0, m = nums.size() + 1;
        int size = nums.size();
        while(start < size && end < size) {
        	while(S < s && end < size) {
        		S += nums[end];
        		end++;
        	}
        	while(S >= s && start <= end) {
        		m = min(m, end - start);
        		S -= nums[start];
        		start += 1;
        	}
        }
        if(m == size + 1)
        	return 0;
        else
        	return m;
    }
};

File: MinimumSizeSubarraySum/MinimumSizeSubarraySum.py
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        start = 0
        end = 0
        tmp_S = 0
        m = len(nums) + 1
        while start < len(nums) and end < len(nums):
            while tmp_S < s and end < len(nums):
                tmp_S += nums[end]
                end += 1
            while tmp_S >= s and start <= end:
                m = min(m, end - start)
                tmp_S -= nums[start]
                start += 1
        if m == len(nums) + 1:
            return 0
        else:
            return m